lib/ivis_opengl/gfx_api_gl.cpp

```diff
@@ -62,6 +92,38 @@ struct gl_texture : public gfx_api::texture
 
 };
 
+struct gl_buffer : public gfx_api::buffer
+{
+	gfx_api::buffer::usage usage;
+	GLuint buffer;
+
+	virtual ~gl_buffer() override
+	{
+		glDeleteBuffers(1, &buffer);
+	}
+
+	gl_buffer(const gfx_api::buffer::usage& usage, const size_t& size, const gfx_api::context::buffer_storage_hint& hint)
+		: usage(usage)
+	{
+		glGenBuffers(1, &buffer);
+		glBindBuffer(to_gl(usage), buffer);
```

Comment:
I'm not sure that `glBindBuffer` should be in the constructor. Don't we want callers to call `bind`? What happens if someone wants to create a buffer, but not bind it yet? This does not provide that option.